hydromt_sfincs/sfincs.py

```diff
@@ -1011,6 +1016,126 @@ def setup_cn_infiltration(self, cn, antecedent_moisture="avg", reproj_method="me
+        refi = dx / 30  # finest resolution of NLCD is ~30m
+        nrcb = int(np.floor(nrmax / refi))  # nr of regular cells in a block
+        nrbn = int(np.ceil(nmax / nrcb))  # nr of blocks in n direction
+        nrbm = int(np.ceil(mmax / nrcb))  # nr of blocks in m direction
+        x_dim, y_dim = self.mask.raster.x_dim, self.mask.raster.y_dim
+
+        # avoid blocks with width or height of 1
+        merge_last_col = False
+        merge_last_row = False
+        if mmax % nrcb == 1:
+            nrbm -= 1
+            merge_last_col = True
+        if nmax % nrcb == 1:
+            nrbn -= 1
+            merge_last_row = True
+
+        ## Loop through blocks
+        ib = -1
+        for ii in range(nrbm):
+            bm0 = ii * nrcb  # Index of first m in block
+            bm1 = min(bm0 + nrcb, mmax)  # last m in block
+            if merge_last_col and ii == (nrbm - 1):
+                bm1 += 1
+
+            for jj in range(nrbn):
+                bn0 = jj * nrcb  # Index of first n in block
+                bn1 = min(bn0 + nrcb, nmax)  # last n in block
+                if merge_last_row and jj == (nrbn - 1):
+                    bn1 += 1
+
+                # Count
+                ib += 1
+                # Get some prints => no idea how to get the logger info
+                print(
+                    f"\nblock {ib + 1}/{nrbn * nrbm} -- "
+                    f"col {bm0}:{bm1-1} | row {bn0}:{bn1-1}"
+                )
+
+                # calculate transform and shape of block at cell and subgrid level
+                da_mask_block = self.mask.isel(
+                    {x_dim: slice(bm0, bm1), y_dim: slice(bn0, bn1)}
+                ).load()
+
+                # Call workflow
+                [
+                    da_smax_block,
+                    da_kr_block,
+                ] = workflows.merge.curvenumber_recovery_determination(
+                    da_landuse, da_HSG, da_Ksat, df_map, da_mask_block
+                )
+
+                # New place in the overall matrix
+                sn, sm = slice(bn0, bn1), slice(bm0, bm1)
+                da_smax[sn, sm] = da_smax_block
+                da_kr[sn, sm] = da_kr_block
+
+        # Done
+        print(" done with determination of values (in blocks)")
+
+        # Specify the effective soil retention (seff)
+        da_seff = da_smax
+        da_seff = da_seff * effective
+        da_seff.attrs.update({"_FillValue": da_smax._FillValue})
+
+        # set grids for seff, smax and kr
+        names = {"smax", "seff", "kr"}
+        for name in names:
+            # Dynamically create the variable name
+            var_name = f"da_{name}"
+
+            # Give metadata to the layer
+            eval(var_name).attrs.update(**self._ATTRS.get(name, {}))
+            self.set_grid(eval(var_name), name=name)  # not sure what this is anymore
+
+            # update config: set maps
+            self.set_config(f"{name}file", f"sfincs.{name}")  # give it to SFINCS
```

Comment:
If possible I suggest to avoid using eval. Here you can avoid it using

```suggestion
        # set grids for seff, smax and kr
        names = ["smax", "seff", "kr"]
        data = [da_smax, da_seff, da_kr]
        for name, da in zip(names, data):
            # Give metadata to the layer and set grid
            da.attrs.update(**self._ATTRS.get(name, {}))
            self.set_grid(da, name=name)

            # update config: set maps
            self.set_config(f"{name}file", f"sfincs.{name}")  # give it to SFINCS
```